src/components/shared/Input/AutoComplete/index.tsx

```diff
@@ -1,12 +1,11 @@
 import React, { useState } from 'react';
-import { AutoComplete as PRAutoComplete  } from 'primereact/autocomplete';
+import { AutoComplete as PRAutoComplete } from 'primereact/autocomplete';
 
-function IAutoComplete() {
-    let [selectedLocation, setLocation] = useState()
+function IAutoComplete({ initialSelectedLocation, onSelectionChange }: { initialSelectedLocation: string | undefined, onSelectionChange: (value: string) => void }) {
+    // let [selectedLocation, setLocation] = useState<string[] | undefined>(initialSelectedLocation)
     let [filteredLocations, setFilteredLocations] = useState<string[] | undefined>()
     const allLocations: string[] = ["Sirsa", "Delhi", "Patiala", "Gurgaon", "Shimla"];
```

Comment:
Constants should be outside the functional component definition. These will also get re-initialized with every re-render.